File: neatformcore/src/main/java/com/nerdstone/neatformcore/domain/model/form/NForm.java
package com.nerdstone.neatformcore.domain.model.form;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.List;

public class NForm implements Serializable {

    @SerializedName("form")
    private String formName;

    @SerializedName("is_multi_step")
    private Boolean isMultiStep;

    @SerializedName("rules_file")
    private String rulesFile;

    @SerializedName("count")
    private int count;

    @SerializedName("steps")
    private List<NFormContent> steps;

    public NForm() {
    }

    public NForm(String formName) {
        this.formName = formName;
    }

    public String getFormName() {
        return formName;
    }

    public void setFormName(String formName) {
        this.formName = formName;
    }

    public Boolean isMultiStep() {
        return isMultiStep;
    }

    public void setMultiStep(Boolean multiStep) {
        isMultiStep = multiStep;
    }

    public int getCount() {
        return count;
    }

    public void setCount(int count) {
        this.count = count;
    }

    public List<NFormContent> getSteps() {
        return steps;
    }

    public void setSteps(List<NFormContent> steps) {
        this.steps = steps;
    }

    public String getRulesFile() {
        return rulesFile;
    }

    public void setRulesFile(String rulesFile) {
        this.rulesFile = rulesFile;
    }
}

File: neatformcore/src/main/java/com/nerdstone/neatformcore/domain/model/form/NForm.kt
package com.nerdstone.neatformcore.domain.model.form

import com.google.gson.annotations.SerializedName

import java.io.Serializable

class NForm : Serializable {

    @SerializedName("form")
    lateinit var formName: String

    @SerializedName("is_multi_step")
    var isMultiStep: Boolean = false

    @SerializedName("rules_file")
    var rulesFile: String? = null

    @SerializedName("count")
    var count: Int = 0

    @SerializedName("steps")
    lateinit var steps: List<NFormContent>

    @SerializedName("meta_data")
    val formMetadata: Map<String, Any>? = null

}
